modules/m4d-implicit-copy-batch/values.yaml

```diff
@@ -12,7 +12,7 @@
 # app_cluster:
 labels: {}
 
-image: "ghcr.io/the-mesh-for-data/mover:latest"
+image: ""
```

Comment:
I think this is a good change. That way the default image is the one that is configured in the MOVER_IMAGE environment variable. Alternatively one can always overwrite this value in the module definition.

Reply:
If I understand correctly, the default behavior is that a webhook defaults image to MOVER_IMAGE. What happens when webhooks are not running, e.g. in tests?

Reply:
Good catch. 

What would be the preference:
1. Omit `mover` section from m4d chart so there will be no default MOVER_IMAGE. Instead, the `image` value in the module `values.yaml` is the default.
2. Default also in reconcile loop. This ensures that the functional behavior is the same even if no webhooks are running (there is still a benefit in running the webhooks because it will block invalid input on submission and avoid changing the spec state to be invalid). In previous projects I used this code as part of the reconcile function:
```go
	// Validate the resource
	switch instance.(type) {
	case webhook.Validator:
		if err := instance.(webhook.Validator).ValidateCreate(); err != nil {
			reqLogger.Error(err, "resource is invalid")
			return r.ManageError(instance, err)
		}
	}

	// Set default values for the resource field
	switch instance.(type) {
	case webhook.Defaulter:
		beforeDefaults := instance.DeepCopyObject()
		instance.(webhook.Defaulter).Default()
		if !equality.Semantic.DeepEqual(beforeDefaults, instance) {
			reqLogger.Info("updating resource after setting defaults")
			if err := r.Client().Update(context.TODO(), instance); err != nil {
				return r.ManageError(instance, err)
			}
			return ctrl.Result{Requeue: true}, nil
		}
	}
```

Reply:
I like the first approach.

Reply:
The first approach would require setting the image in each BatchTransfer/StreamTransfer when using directly and not as a module. Do we care? @froesef it's your call what to do.

Reply:
I agree with removing the `MOVER_IMAGE` parameter. It's a relic from the times when the motion components were separate and it was easier for testing. What we could also do is define a default in the CRD definition (controller-gen annotations). Should I do a PR for this?

Reply:
> What we could also do is define a default in the CRD definition (controller-gen annotations).

You can't. Defaulting requires apiextensions.k8s.io/v1 so #501 must be fixed first. 
Defaulting is also not enabled by default on Kubernetes 1.16 (which is the minimal version we support).
I just had the same discussion with Tomer.

Reply:
With that in mind feel free to prepare a separate PR or take over this one.